src/libraries/System.Runtime/tests/System/BufferTests.cs

```diff
@@ -91,6 +91,10 @@ public static void ByteLength_Invalid()
         [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 7, 0x89)]
         public static void GetByte(Array array, int index, int expected)
         {
+            if (!BitConverter.IsLittleEndian)
+            {
+                index = index ^ 3;
```

Comment:
What is this doing?

Reply:
The input is an array of uint, which is accessed via a byte index.  To compensate for byte order, this patch simply updates the index to point to the "swapped" position (0 -> 3, 1 -> 2, 2 -> 1, 3 -> 0, 4 -> 7, 5 -> 6, 6 -> 5, 7 -> 4 etc.).  This index transformation can be implemented via the ^ 3 operation.